dev-python/pipenv/pipenv-2024.4.1.ebuild

```diff
@@ -0,0 +1,127 @@
+# in the pipenv source. Attempts to simply bump the version of the
+# package without checking that it works is likely to fail
+# The vendored packages should eventually all be removed
+# see: https://bugs.gentoo.org/717666
+src_prepare() {
+	sed --in-place -e "s/import click, plette, tomlkit/import click\n\import tomlkit\nfrom pipenv.vendor import plette/g" pipenv/project.py || die "Failed patching pipenv/project.py"
+
+	local pkgName
+	local jobs=$(makeopts_jobs)
+	local packages=( cerberus colorama click click_didyoumean dotenv dparse markupsafe \
+					 pexpect pep517 pipdeptree plette ptyprocess pydantic pyparsing pythonfinder \
+					 requests urllib3 shellingham tomli tomlkit importlib_metadata )
+	for pkgName in ${packages[@]}; do
+		find ./ -type f -print0 | \
+			xargs --max-procs="${jobs}" --null \
+			sed --in-place \
+				-e "s/from pipenv.vendor import ${pkgName}/import ${pkgName}/g" \
+				-e "s/from pipenv.vendor.${pkgName}\(.*\) import \(\w*\)/from ${pkgName}\1 import \2/g"\
+				-e "s/import pipenv.vendor.${pkgName} as ${pkgName}/import ${pkgName}/g" \
+				-e "s/from .vendor import ${pkgName}/import ${pkgName}/g" \
+		        -e "s/from .vendor.${pkgName}/from ${pkgName}/g" || die "Failed to sed for ${pkgName}"
+	done
+
+	distutils-r1_src_prepare
+
+	# remove vendored versions
+	for pkgName in ${packages[@]}; do
+		find  ./pipenv/vendor -regextype posix-extended -regex ".*${pkgName}$" -prune -exec rm -rvf {} + || die
+		# package names can be foo-bar, their module will be however foo_bar
+		find  ./pipenv/vendor -regextype posix-extended -regex ".*${pkgName/_/-}" -prune -exec rm -rvf {} + || die
+	done
+
+	find  ./pipenv/vendor -regextype posix-extended -regex '.*cached[_-]property.*' -prune -exec rm -rvf {} + || die
+
+	find ./ -type f -print0 | \
+		xargs --max-procs="${jobs}" --null \
+		sed --in-place \
+			-e "s/from pipenv\.vendor import plette, toml, tomlkit, vistir/from pipenv\.vendor import plette, toml, vistir\\nimport tomlkit/g"
+
+	# remove python ruaml yaml
+	sed --in-place -e \
+		"s/from pipenv\.vendor\.ruamel\.yaml import YAML/from ruamel\.yaml import YAML/g" \
+		pipenv/patched/safety/util.py || die "Failed sed in ruaml-yaml"
+	sed --in-place -e \
+		"s/from pipenv\.vendor\.ruamel\.yaml\.error import MarkedYAMLError/from ruamel\.yaml\.error import MarkedYAMLError/g" \
+		pipenv/patched/safety/util.py || die "Failed sed in ruamel-yaml"
+
+	rm -vR pipenv/vendor/ruamel || die "Failed removing ruamel-yaml from vendor"
+
+	for fname in Makefile README.md ruamel.*.LICENSE vendor.txt; do
+		rm -v pipenv/vendor/$fname || die "Failed removing pipenv/vendor/${fname}"
+	done
+
+	sed --in-place -e "s/pipenv.vendor.pythonfinder.utils.get_python_version/pythonfinder.utils.get_python_version/g" tests/unit/test_utils.py || die "Failed patching tests"
+
+	rm -Rfv pipenv/vendor || die "Could not vendor"
+	rm -Rfv examples || die "Could not remove examples"
+	rm -Rfv docs || die "Could not remove docs"
```

Comment:
We generally don't rely on `-f` **rm** switch - you ought to know what you're removing ;)

Reply:
I addressed this comment in the new pipenv-2025.0.2.ebuild.